runtime/engine/src/test/java/io/aklivity/zilla/runtime/engine/test/internal/catalog/TestCatalogHandler.java

```diff
@@ -15,23 +15,61 @@
  */
 package io.aklivity.zilla.runtime.engine.test.internal.catalog;
 
+import java.nio.ByteOrder;
+
+import org.agrona.MutableDirectBuffer;
+import org.agrona.concurrent.UnsafeBuffer;
+
 import io.aklivity.zilla.runtime.engine.catalog.CatalogHandler;
 import io.aklivity.zilla.runtime.engine.test.internal.catalog.config.TestCatalogOptionsConfig;
+import io.aklivity.zilla.runtime.engine.validator.function.ValueConsumer;
 
 public class TestCatalogHandler implements CatalogHandler
 {
+    private static final int MAX_PADDING_LEN = 10;
+    private static final byte MAGIC_BYTE = 0x0;
+    private static final int ENRICHED_LENGTH = 5;
+
     private final String schema;
+    private final MutableDirectBuffer prefixRO;
+    private final int id;
+    private final boolean embed;
+    private final boolean exclude;
 
     public TestCatalogHandler(
-        TestCatalogOptionsConfig options)
+        TestCatalogOptionsConfig config)
+    {
+        this.schema = config.schema;
+        this.prefixRO = new UnsafeBuffer(new byte[5]);
+        this.id = config.id;
+        this.embed = config.embed;
+        this.exclude = config.exclude;
+    }
+
+    @Override
+    public int enrich(
+        int schemaId,
+        ValueConsumer next)
```

Comment:
What is the intention of the return value of `enrich`?

It seems like it is trying to say how much of the input should be ignored, while also potentially writing something to `next`.

So in the case of `embed` where we are adding the prefix bytes, the `PREFIX_LENGTH` would already be passed to `next`, so perhaps this case should return `0`?

And when we are doing `exclude`, meaning we are stripping off the `PREFIX_LENGTH` bytes, we would not call `next` so we need to return `PREFIX_LENGTH` to indicate how many bytes we skipped over.

The implementation below is returning `PREFIX_LENGTH` for both cases, is it intentional?